autoscale/src/main/java/com/sequenceiq/periscope/service/AltusMachineUserService.java

```diff
@@ -0,0 +1,98 @@
+                    grpcUmsClient.assignResourceRole(machineUserCrn, environmentCrn,
+                            roleCrnGenerator.getBuiltInEnvironmentUserResourceRoleCrn(), MDCUtils.getRequestId());
+                    LOGGER.info("Assigned resourcerole '{}' for  machineUserCrn '{}' for environment '{}'",
+                            roleCrnGenerator.getBuiltInEnvironmentUserResourceRoleCrn(), machineUserCrn, environmentCrn);
+                }
+                syncEnvironment(cluster.getClusterPertain().getTenant(), machineUserCrn, environmentCrn, Optional.empty());
+                clusterService.setMachineUserCrn(cluster.getId(), machineUserCrn);
+            }
+        } catch (Exception ex) {
+            LOGGER.warn("Error initializing machineUserCrn for cluster '{}' yarn polling", cluster.getStackCrn(), ex);
+        }
+    }
+
+    @Retryable(value = Exception.class, maxAttempts = 5, backoff = @Backoff(delay = 10000))
+    public void deleteMachineUserForEnvironment(String accountId, String machineUserCrn, String environmentCrn) {
+        if (environmentCrn != null && machineUserCrn != null) {
+            MachineUser machineUser = getOrCreateAutoscaleMachineUser(environmentCrn, accountId);
+            grpcUmsClient.deleteMachineUser(machineUser.getCrn(), ThreadBasedUserCrnProvider.INTERNAL_ACTOR_CRN, accountId, MDCUtils.getRequestId());
+            syncEnvironment(accountId, machineUserCrn, environmentCrn, Optional.of(machineUser.getWorkloadUsername()));
+            LOGGER.info("Deleted MachineUser for machineUserCrn '{}', environment '{}'", machineUserCrn, environmentCrn);
+        }
+    }
+
+    private MachineUser getOrCreateAutoscaleMachineUser(String environmentCrn, String accountId) {
+        //Idempotent api retrieves machine user or creates if missing.
+        String autoscaleMachineUserName = String.format(AUTOSCALE_MACHINE_USER_NAME_PATTERN, Crn.fromString(environmentCrn).getResource());
+        MachineUser machineUser = grpcUmsClient.getOrCreateMachineUserWithoutAccessKey(autoscaleMachineUserName, accountId, MDCUtils.getRequestId());
+        LOGGER.info("Retrieved machineUser '{}' for machineUserName '{}' ", machineUser, autoscaleMachineUserName);
+        return machineUser;
+    }
+
+    private void syncEnvironment(String accountId, String machineUserCrn, String environmentCrn, Optional<String> deletedWorkloadUserName) {
+        ThreadBasedUserCrnProvider.doAsInternalActor(() -> {
+            SynchronizeAllUsersRequest request = new SynchronizeAllUsersRequest();
+            request.setAccountId(accountId);
+            request.setEnvironments(Set.of(environmentCrn));
+            request.setMachineUsers(Set.of(machineUserCrn));
+            if (deletedWorkloadUserName.isPresent()) {
+                request.setDeletedWorkloadUsers(Set.of(deletedWorkloadUserName.get()));
+            }
+            userV1Endpoint.synchronizeAllUsers(request);
```

Comment:
are you sure we need this? I would ask @handavid or @gitmfox if there is automatic sync implemented or not, because I would avoid to introduce a new service-to-service relation (autoscale - freeipa), if possible.

Reply:
I see there were already another thread about this, although if there is no high priority on this task, I would seek for solution/wait for solution which removes the necessity of this sync, since this will require addition configuration also in CSI repo before merging this change.

Reply:
Has automatic sync been implemented? A jira has been filed to track the removal of this explicit userSync: [CB-14554](https://jira.cloudera.com/browse/CB-14554)

Reply:
@aakashshankar it seems that is an old ticket, please check if manual sync is still needed or not